### PR TITLE
JS: improve join-order for Configuration::barrierGuardBlocksEdge

### PR DESCRIPTION
**Pull request by erik-krogh**
The predicate went from ~40s to ~100ms when running `TaintedPath.ql` on `ace`. 

The goal of my reordering was mostly to get `barrierGuardIsRelevant` earlier in the join-order.

### CODE REVIEW

**Comment by erik-krogh** [comment]
[Evaluation looks OK](https://github.com/dsp-testing/erik-krogh-BCQS/tree/auto/data/workflow/barrierperf-eval-1/reports).

File: javascript/ql/src/semmle/javascript/dataflow/Configuration.qll
Comment (tausbn) [inline javascript/ql/src/semmle/javascript/dataflow/Configuration.qll:494]: Should this perhaps be `pragma[noinline]` to prevent the compiler from inlining the predicate and discovering the bad join order again?

**Comment by asgerf** [comment]
Can you hold off on this until after the Redux PR is merged, or run an evaluation of it based on that? It changes the `dominates` predicate in a significant way and I'd like to avoid a semantic merge conflict.

**Comment by asgerf** [comment]
> Can you hold off on this until after the Redux PR is merged

OK that was a bad suggestion given that it just hit the BDD node limit again. Nevermind.

File: javascript/ql/src/semmle/javascript/dataflow/Configuration.qll
Comment (erik-krogh) [reply javascript/ql/src/semmle/javascript/dataflow/Configuration.qll:494]: Maybe. 
I prefer not to. 
If the compiler is changed such that it inlines the predicate, it might also be changed such that it discovers a better join order. 
If we use a `pragma`, we force the compiler to use the current join order, even if something better is possible in the future.

**Review by asgerf — APPROVED** [review]

**Pull request merged**
